mlflow/gateway/providers/anthropic.py

```diff
@@ -0,0 +1,104 @@
+    SUPPORTED_ROUTES = "completions"
+
+    async def _request(self, path: str, payload: Dict[str, Any]):
+        import aiohttp
+
+        config = self.config.model.config
+        token = config["anthropic_api_key"]
+        headers = {"x-api-key": token}
+        async with aiohttp.ClientSession(headers=headers) as session:
+            url = "/".join([config["anthropic_api_base"].rstrip("/"), path.lstrip("/")])
+            async with session.post(url, json=payload) as response:
+                js = await response.json()
+                try:
+                    response.raise_for_status()
+                except aiohttp.ClientResponseError as e:
+                    detail = js.get("error", {}).get("message", e.message)
+                    raise HTTPException(status_code=e.status, detail=detail)
+                return js
+
+    @staticmethod
+    def _make_payload(payload: Dict[str, Any], mapping: Dict[str, str]):
+        payload = payload.copy()
+        for k1, k2 in mapping.items():
+            if v := payload.pop(k1, None):
+                payload[k2] = v
+        return {k: v for k, v in payload.items() if v is not None and v != []}
+
+    async def completions(self, payload: completions.RequestPayload) -> completions.ResponsePayload:
+        payload = jsonable_encoder(payload)
+        if "top_p" in payload:
+            raise HTTPException(
+                status_code=400,
+                detail="Cannot set both 'temperature' and 'top_p' parameters. "
+                "Please use only the temperature parameter for your query.",
+            )
+        if payload["max_tokens"] is None:
+            raise HTTPException(
+                status_code=400,
+                detail="You must set an integer value for 'max_tokens' for the Anthropic provider "
+                "that provides the upper bound on the returned token count.",
+            )
+        if payload.get("stream", None) == "true":
+            raise HTTPException(
+                status_code=400,
+                detail="Setting the 'stream' parameter to 'true' is not supported with the MLflow "
+                "Gateway.",
+            )
+
+        payload = AnthropicProvider._make_payload(
+            payload, {"max_tokens_to_sample": "max_tokens", "stop_sequences": "stop"}
+        )
+
+        payload["prompt"] = f"\n\nHuman: {payload['prompt']}\n\nAssistant:"
+
+        resp = await self._request("complete", {"model": self.config.model.name, **payload})
+
+        # Example response:
+        # {
+        #     "completion": " Here is a basic overview of how a car works:\n\n1. The engine. "
+        #                   "The engine is the power source that makes the car move."
+        #     "stop_reason": "max_tokens",
+        #     "model": "claude-instant-1.1",
+        #     "truncated": False,
+        #     "stop": None,
+        #     "log_id": "dee173f87ddf1357da639dee3c38d833",
+        #     "exception": None,
+        # }
```

Comment:
Does Anthropic have an API reference like the one OpenAI has?

Reply:
it does, but not as fully-featured. The examples do not match the actual response fields that are returned when querying their REST APIs. I updated with a link to their docs and matching the example to the fields that are in the docs.